crates/biome_css_analyze/src/lint/nursery/use_sorted_properties.rs

```diff
@@ -0,0 +1,387 @@
+            },
+        ));
+    }
+
+    fn action(ctx: &RuleContext<Self>, state: &Self::State) -> Option<CssRuleAction> {
+        if state.is_unsafe_to_sort || state.sorted_properties.is_sorted() {
+            return None;
+        }
+
+        let mut mutation = ctx.root().begin();
+        mutation.replace_node_discard_trivia(
+            state.block.items(),
+            state
+                .sorted_properties
+                .as_sorted(&state.original_properties)?,
+        );
+
+        return Some(RuleAction::<CssLanguage>::new(
+            ActionCategory::QuickFix(Cow::Borrowed("")),
+            ctx.metadata().applicability(),
+            markup! { "Sort these properties" }.to_owned(),
+            mutation,
+        ));
+    }
+}
+
+/// "Recess order" is a semantic ordering that mimics the behavior of [stylelint-config-recess-order](https://github.com/stormwarning/stylelint-config-recess-order).
+///
+/// Which in turn mimics the behavior of twitter's [RECESS](https://github.com/twitter-archive/recess/blob/29bccc870b7b4ccaa0a138e504caf608a6606b59/lib/lint/strict-property-order.js).
+pub struct RecessOrderProperties(pub BTreeSet<SortInfo>);
+
+impl RecessOrderProperties {
+    pub fn new(items: &[AnyCssDeclarationOrRule]) -> Self {
+        RecessOrderProperties(
+            items
+                .iter()
+                .enumerate()
+                .map(|(position, node)| {
+                    let mut sort_info = SortInfo::from(node);
+                    sort_info.original_position = position;
+                    sort_info
+                })
+                .collect::<BTreeSet<SortInfo>>(),
+        )
+    }
+
+    pub fn is_sorted(&self) -> bool {
+        // The list is sorted if the original_position field equals the actual position for every item
+        self.0
+            .iter()
+            .enumerate()
+            .all(|(position, item)| position == item.original_position)
+    }
+
+    pub fn as_sorted(
+        &self,
+        original_items: &[AnyCssDeclarationOrRule],
+    ) -> Option<CssDeclarationOrRuleList> {
+        CssDeclarationOrRuleList::cast(SyntaxNode::new_detached(
+            CssSyntaxKind::CSS_DECLARATION_OR_RULE_LIST,
+            self.0.iter().map(|sort_info| {
+                original_items
+                    .get(sort_info.original_position)
+                    .map(|node| NodeOrToken::Node(node.clone().into_syntax()))
+            }),
+        ))
+    }
+}
+
+fn css_identifier_to_prop_text(ident: &CssIdentifier) -> Option<TokenText> {
+    let tok = ident.value_token().ok()?;
+    Some(tok.token_text_trimmed())
+}
+
+fn css_declaration_to_prop_text(
+    decl_with_semicolon: &CssDeclarationWithSemicolon,
+) -> Option<TokenText> {
+    decl_with_semicolon
+        .declaration()
+        .ok()?
+        .property()
+        .ok()?
+        .as_css_generic_property()?
+        .name()
+        .ok()
+        .and_then(|prop_name| -> Option<TokenText> {
```

Comment:
```suggestion
        .and_then(|prop_name| {
```

No need to add a type, Rust compiler is very good at inferring types. Usually, you want to add types only when the compiler tells you to.